benchmarks/benchmark/tools/locust-load-inference/locust-docker/locust-tasks/custom_metric_aggregator.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+class TokenMetricCollector:
+    def __init__(self):
+        self.tokens_sent = []
+        self.tokens_received = []
+        self.test_time = []
+        self.success_count = 0
+        self.failure_count = 0
+        
+
+    def add_metric(self, sent, received, test_time, request_succesful_bool):
+        if request_succesful_bool==1:
+            self.tokens_sent.append(sent)
+            self.tokens_received.append(received)
+            self.test_time.append(test_time)
+            self.success_count += 1
+        else:
+            self.failure_count += 1
+            
+
+    def calculate_average_tokens(self):
+        avg_sent = sum(self.tokens_sent) / len(self.tokens_sent) if self.tokens_sent else 0
+        avg_received = sum(self.tokens_received) / len(self.tokens_received) if self.tokens_received else 0
+        avg_test_time = sum(self.test_time) / len(self.test_time) if self.tokens_sent else 0
+        
+        return avg_sent, avg_received, avg_test_time
+
+    def write_to_csv(self, file_path='custom_metrics.csv'):
+        import csv
+        avg_sent, avg_received, avg_test_time = self.calculate_average_tokens()
+        with open(file_path, mode='w', newline='') as file:
+            writer = csv.writer(file)
+            writer.writerow(['Metric', 'Average Value'])
+            writer.writerow(['# of Successful Req', self.success_count])
+            writer.writerow(['# of Failed Req', self.failure_count])
+            writer.writerow(['Avg Tokens Sent Per Req', avg_sent])
```

Comment:
Thanks for adding these. There are a couple of metrics that are more important and would be good to add:
1. Throughput in output tokens / sec - includes all requests in the run.
2.  Avg request latency in milliseconds per output token

It is ok to add them in a separate change if you'd like.

Reply:
yes will do in a follow up PR